app_config_b2c.py

```diff
@@ -1,45 +1,37 @@
 import os
 
-b2c_tenant = "fabrikamb2c"
-signupsignin_user_flow = "B2C_1_signupsignin1"
-editprofile_user_flow = "B2C_1_profileediting1"
+b2c_tenant = os.getenv('TENANT_ID')
+signupsignin_user_flow = os.getenv('SIGNUPSIGNIN_USER_FLOW')
+editprofile_user_flow = os.getenv('EDITPROFILE_USER_FLOW')
 
-resetpassword_user_flow = "B2C_1_passwordreset1"  # Note: Legacy setting.
-    # If you are using the new
-    # "Recommended user flow" (https://docs.microsoft.com/en-us/azure/active-directory-b2c/user-flow-versions),
-    # you can remove the resetpassword_user_flow and the B2C_RESET_PASSWORD_AUTHORITY settings from this file.
+resetpassword_user_flow = os.getenv('RESETPASSWORD_USER_FLOW')  # Note: Legacy setting.
+# If you are using the new
+# "Recommended user flow" (https://docs.microsoft.com/en-us/azure/active-directory-b2c/user-flow-versions),
+# you can remove the resetpassword_user_flow and the B2C_RESET_PASSWORD_AUTHORITY settings from this file.
 
 authority_template = "https://{tenant}.b2clogin.com/{tenant}.onmicrosoft.com/{user_flow}"
```

Comment:
Ahh, only when attempting changing this line, I remembered why we intentionally chose to use the short tenant name (i.e. fabrikam) in B2C configuration. Its authority template would require TWO occurrences of the short tenant name.

`https://fabrikam.b2clogin.com/fabrikam.onmicrosoft.com/{user_flow}`

This does not play well with the TENANT_ID = "fabrikam.onmicrosoft.com" input.

Perhaps, let's change line 3 into this?

```python
b2c_tenant = os.getenv('TENANT_ID').split(".")[0]  # Parse and use the "contoso" segment
```

Reply:
Ahh.. Okay so for B2C, it actually can't be the Tenant GUID, right, it can only be the name? We can use TENANT_NAME for b2c and TENANT_ID for standard- the env variables don't need to match.